### PR TITLE
chore: extract common functions to utils pkg

### PR DESCRIPTION
**Pull request by jrsmroz**
**What this PR does / why we need it**:

Extracts common function in gateway utils that makes public functions shorter and easier to read.
Extracted from https://github.com/Kong/gateway-operator/pull/70 that is going to use those functions too.

**Which issue this PR fixes**

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

**Comment by mlavacca** [comment]
This looks good to me. I'm wondering if we can make this function even more generic, something like `newManagedByObjectListSelector(obj metav1.Object)`, that we could use also to retrieve objects owned by controlplanes and dataplanes (we have this copy-pasta code also in `ListDeploymentsForOwner` and `ListServicesForOwner`. WDYT?

**Review by mlavacca — COMMENTED** [review]
This looks good to me. I'm wondering if we can make this function even more generic, something like `newManagedByObjectListSelector(obj metav1.Object)`, that we could use also to retrieve objects owned by controlplanes and dataplanes (we have this copy-pasta code also in `ListDeploymentsForOwner` and `ListServicesForOwner`). WDYT?

**Comment by jrsmroz** [comment]
> This looks good to me. I'm wondering if we can make this function even more generic, something like `newManagedByObjectListSelector(obj metav1.Object)`, that we could use also to retrieve objects owned by controlplanes and dataplanes (we have this copy-pasta code also in `ListDeploymentsForOwner` and `ListServicesForOwner`). WDYT?

I will take a look and see what I can do about that.

File: internal/utils/kubernetes/lists.go [hunk 1]
```diff
@@ -68,37 +53,49 @@ func ListServicesForOwner(
...
 
 	return services, nil
 }
+
+// NewListSelectorOption is a helper function to create a ListSelectorOption
+// from a namespace, label key and label values.
+func NewListSelectorOption(
+	namespace string,
+	labelKey string,
+	op selection.Operator,
```
[the remark this answers is not in the document]
Comment (jrsmroz) [reply internal/utils/kubernetes/lists.go:82]: Oh, good catch.

**Comment by jrsmroz** [comment]
Actually when I think about that, we might not need to create ListOptions by hand.
This should be enough:
```go
c.List(ctx, serviceList, client.InNamespace(namespace), client.MatchingLabels{requiredLabel: requiredValue})
```
Gonna test that.

File: internal/utils/gateway/ownerrefs.go [hunk 2]
```diff
@@ -31,33 +29,23 @@ func ListDataPlanesForGateway(
...
 
-	listOptions := &client.ListOptions{
-		Namespace:     gateway.Namespace,
-		LabelSelector: selector,
-	}
+	err := c.List(
+		ctx,
+		dataplaneList,
+		client.InNamespace(gateway.Namespace),
+		client.MatchingLabels{consts.GatewayOperatorControlledLabel: consts.GatewayManagedLabelValue},
```
[the remark this answers is not in the document]
Comment (jrsmroz) [reply internal/utils/gateway/ownerrefs.go:38]: Agree. That's really not perfect.
```go
const (
	// GatewayOperatorControlledLabel is the label that is used for objects which
	// were created by this operator.
	GatewayOperatorControlledLabel = "konghq.com/gateway-operator"

	// DataPlaneManagedLabelValue indicates that an object's lifecycle is managed
	// by the dataplane controller.
	DataPlaneManagedLabelValue = "dataplane"

	// ControlPlaneManagedLabelValue indicates that an object's lifecycle is managed
	// by the controlplane controller.
	ControlPlaneManagedLabelValue = "controlplane"

	// GatewayManagedLabelValue indicates that the object's lifecycle is managed by
	// the gateway controller.
	GatewayManagedLabelValue = "gateway"
)
```

It needs renaming and usage check. I don't wanna think about that right now.

**Pull request merged**